subprojects/base-services/src/main/java/org/gradle/internal/file/DefaultFileHierarchySet.java

```diff
@@ -132,7 +132,7 @@ public Node(String prefix, List<Node> children) {
         Node plus(String path) {
             int maxPos = Math.min(prefix.length(), path.length());
             int prefixLen = sizeOfCommonPrefix(path, 0);
-            if (prefixLen == maxPos) {
+            if (prefixLen == maxPos && prefixLen != 0) {
```

Comment:
Don't you need `prefixLen == maxPos && prefixLen == 0` for querying the root of the file system?